cypress/integration/brewery/ScenarioSharing.js

```diff
@@ -1,40 +1,211 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { Login, Scenarios, ScenarioParameters, ScenarioManager } from '../../commons/actions';
+import { RolesEdition } from '../../commons/actions/generic/RolesEdition';
 import { stub } from '../../commons/services/stubbing';
 import { setup } from '../../commons/utils/setup';
-
-// const SCENARIO_DATASET = DATASET.BREWERY_ADT;
-// const SCENARIO_RUN_TEMPLATE = RUN_TEMPLATE.BASIC_TYPES;
+import { USER_EXAMPLE, USERS_LIST } from '../../fixtures/stubbing/default';
+import {
+  UNSHARED_SCENARIOS_LIST,
+  SHARED_SCENARIOS_LIST,
+  WORKSPACES_LIST,
+} from '../../fixtures/stubbing/ScenarioSharing';
+import { ROLES, PERMISSIONS } from '../../commons/constants/generic/TestConstants';
 
 describe('Check workspace permissions for admin', () => {
-  before(() => {
+  const stubbingResourcesArray = [
+    { user: USER_EXAMPLE, scenarioList: UNSHARED_SCENARIOS_LIST },
+    { user: USER_EXAMPLE, scenarioList: UNSHARED_SCENARIOS_LIST },
+    { user: USER_EXAMPLE, scenarioList: UNSHARED_SCENARIOS_LIST },
+    { user: USERS_LIST[1], scenarioList: SHARED_SCENARIOS_LIST },
+    { user: USERS_LIST[2], scenarioList: SHARED_SCENARIOS_LIST },
+    { user: USERS_LIST[3], scenarioList: SHARED_SCENARIOS_LIST },
+  ];
+
+  let stubbingResourcesArrayIndex = 0;
+
+  beforeEach(() => {
     setup.initCypressAndStubbing();
     stub.start();
+    stub.setFakeUser(stubbingResourcesArray[stubbingResourcesArrayIndex].user);
+    stub.setWorkspaces(WORKSPACES_LIST);
+    stub.setFakeWorkspaceId('W-stbbdbwsx1');
+    stub.setScenarios(stubbingResourcesArray[stubbingResourcesArrayIndex].scenarioList);
     Login.login();
   });
 
-  beforeEach(() => {
-    Login.relogin();
-  });
-
-  after(() => {
+  afterEach(() => {
     stub.stop();
+    ++stubbingResourcesArrayIndex;
   });
 
   it('can create, edit, launch, validate, share, rename & delete a scenario', () => {
     // TODO: check buttons for actions described above are not disabled for an admin user
+    Scenarios.getScenarioValidateButton().should('be.visible').should('not.be.disabled');
+    Scenarios.getScenarioRejectButton().should('be.visible').should('not.be.disabled');
+    Scenarios.getScenarioCreationButton().should('be.visible').should('not.be.disabled');
+    ScenarioParameters.getParametersEditButton().should('be.visible').should('not.be.disabled');
+    ScenarioParameters.getLaunchButton().should('be.visible').should('not.be.disabled');
+    RolesEdition.getShareScenarioButton().should('be.visible').should('not.be.disabled').click();
+    RolesEdition.getShareScenarioDialogAgentsSelect().should('be.visible').should('not.be.disabled');
+    RolesEdition.getShareScenarioDialogFirstCancelButton().click();
+    ScenarioManager.switchToScenarioManager();
+
+    ScenarioManager.writeInFilter(UNSHARED_SCENARIOS_LIST[0].name);
+    ScenarioManager.getDeleteScenarioButton().should('be.visible').should('not.be.disabled');
```

Comment:
We should probably create an action to first get the scenario card (or accordion) and then the delete button inside it. If we don't provide the id or name of the scenario whose delete button we want to check, we may detect the delete button of another scenario (even if the scenario name is written in the filter, other scenarios could have the same pattern with a longer name)
This new action could use `getScenarioAccordion(scenarioId)`

Reply:
In fact, this function already exists. We just don't use it yet.